### PR TITLE
Update struct

### PR DESCRIPTION
**Pull request by ghost**
In this branch I try to implement new structure for ccontrollers classes and add to scout shoot target point ability. Also create test level for some gameplay.

Please check is all work well.

### CODE REVIEW

**Comment by ghost** [comment]
@alexnodejs @eugene-d Please check.

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,6 @@ sysinfo.txt
 
 # Other stuff
 .DS_Store
+
+# Third party files folder
+Assets/Third_party_files
```
Comment (alexnodejs) [inline .gitignore]: @zerocool0686  It means we need empty line at the end of file.

**Comment by ghost** [comment]
Ok, I'm update gitignore file. @alexnodejs

**Comment by eugene-d** [comment]
@zerocool0686 @alexnodejs works well!

**Pull request merged**